extension/src/popup/components/accountHistory/HistoryItem/index.tsx

```diff
@@ -397,17 +397,26 @@ export const HistoryItem = ({
           setIconComponent(
             <Icon.ArrowUp className="HistoryItem__icon--sent" />,
           );
+          setIsLoading(true);
 
-          if (!tokenKey) {
-            // TODO: attempt to fetch token details, not stored
-            setRowText(operationString);
-            setTxDetails((_state) => ({
-              ..._state,
-              headerTitle: translations("Transaction"),
-              operationText: operationString,
-            }));
-          } else {
-            const { token, decimals } = balances[tokenKey] as TokenBalance;
+          try {
```

Comment:
Now that sending transfers from payments is better supported, it became more obvious that we can't rely on client side cached token details for this path because we more often don't have them(user has a classic balance in Freighter but send to C address through SAC transfer for example). This update just fetches token details for every row that is a transfer inline to avoid the inconsistencies.